runelite-client/src/main/java/net/runelite/client/plugins/herbsack/HerbSackPlugin.java

```diff
@@ -0,0 +1,300 @@
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.ItemContainerChanged;
+import net.runelite.api.events.MenuOptionClicked;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.EventBus;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.infobox.Counter;
+import net.runelite.client.ui.overlay.infobox.InfoBox;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+@Slf4j
+@PluginDescriptor(
+	name = "Herb Sack",
+	description = "Enable the tracking of herbs in the herb sack",
+	tags = {"overlay", "herb", "sack", "bag", "storage"}
+)
+@Singleton
+public class HerbSackPlugin extends Plugin
+{
+	private static final int HERB_BAG_NONE = -1;
+	private static final int HERB_BAG_FILL = 0;
+	private static final int HERB_BAG_EMPTY_INVENTORY = 1;
+	private static final int HERB_BAG_EMPTY_BANK = 2;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private HerbSackOverlay herbSackOverlay;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private EventBus eventBus;
+
+	@Inject
+	private HerbSackConfig config;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	private HashMap<Integer, Counter> infoBoxes;
+	@Getter
+	private HerbSack herbSack;
+	@Getter
+	private boolean showOverlay;
+	@Getter
+	private boolean herbSackInInventory;
+	private boolean showInfoBoxes;
+	private int lastInteraction;
+	private boolean bankChanged;
+	private boolean inventoryChanged;
+	private boolean infoBoxesAdded;
+	private List<HerbCount> lastBankContents;
+	private List<HerbCount> lastInventoryContents;
+
+	@Provides
+	HerbSackConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(HerbSackConfig.class);
+	}
+
+	@Override
+	protected void startUp()
+	{
+		infoBoxesAdded = false;
+		initInfoBoxes();
+		updateConfig();
+		addInfoBoxes();
+		lastInteraction = HERB_BAG_NONE;
+		herbSack = new HerbSack();
+		overlayManager.add(herbSackOverlay);
+		addSubscriptions();
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		eventBus.unregister(this);
+		overlayManager.remove(herbSackOverlay);
+		removeInfoBoxes();
+	}
+
+	private void addSubscriptions()
+	{
+		eventBus.subscribe(MenuOptionClicked.class, this, this::onMenuOptionClicked);
+		eventBus.subscribe(ItemContainerChanged.class, this, this::onItemContainerChanged);
+		eventBus.subscribe(ConfigChanged.class, this, this::onConfigChanged);
+	}
+
+	private void onConfigChanged(ConfigChanged configChanged)
+	{
+		if (!configChanged.getGroup().equals("herbSack"))
+		{
+			return;
+		}
+		updateConfig();
+	}
+
+	private void onItemContainerChanged(ItemContainerChanged itemContainerChanged)
```

Comment:
Why are you doing this no matter what interaction there has been with the herb sack? Say you deposit 10 items into your bank, with or without a herb sack in your inventory, you're iterating over all items in your bank 10 times?